ports/esp8266/machine_uart.c

```diff
@@ -300,7 +301,7 @@ STATIC mp_uint_t mp_machine_uart_ioctl(mp_obj_t self_in, mp_uint_t request, uint
             if (mp_machine_uart_txdone(self)) {
                 return 0;
             }
-            MICROPY_EVENT_POLL_HOOK
+            mp_event_wait_indefinite();
```

Comment:
This should be `mp_event_wait_ms(1)`, since the loop checks for a timeout.

Reply:
Oops, good catch! (I think they're equivalent on this port, but good to use the correct function anyhow.)